test/Microsoft.AspNetCore.Razor.Evolution.Test/RazorSyntaxTreeTest.cs

```diff
@@ -20,5 +20,21 @@ public void Parse_CanParseEmptyDocument()
             Assert.NotNull(syntaxTree);
             Assert.Empty(syntaxTree.Diagnostics);
         }
+
+        [Fact]
+        public void Parse_Persists_FilePath()
+        {
+            // Arrange
+            var filePath = "test.cshtml";
+            var source = TestRazorSourceDocument.Create("@Hello", fileName: filePath);
+
+            // Act
+            var syntaxTree = RazorSyntaxTree.Parse(source);
+
+            // Assert
+            Assert.Empty(syntaxTree.Diagnostics);
+            Assert.NotNull(syntaxTree);
+            Assert.All(syntaxTree.Root.Children, node => Assert.Equal(filePath, node.Start.FilePath));
```

Comment:
Isn't there like one child in the tree? Could we have a more complex graph /w nested children and verify it's persisted in all of those?